gae_serve/README.md

```diff
@@ -0,0 +1,78 @@
+
+## Introduction
+
+This is a simple sample that shows how to:
+
+- Create a GAE service that loads a pickled sci-kit learn model from GCS, and use it to serve prediction requests through Cloud Endpoints.
+
+The benefits of this configuration include:
+
+1. GAE's autoscaling and load balancing.
+1. Cloud Endpoints' monitoring and access control.
+
+
+## Requirements
+
+Python.
+
+GCP SDK.  gsutil.
+
+A GCP project that has GAE, GCS, and Cloud Endpoints enabled.
+
+
+## Setup
+
+1. `git clone https://github.com/GoogleCloudPlatform/ml-on-gcp`
+
+1. `cd ml-on-gcp`
+
+1. `pip install -r requirements.txt`
+
+1. The name `modelserve` is tentative which can be changed to something else, but you need to change all occurrences consistently.
+
+1. Update `modelserve.yaml`:  Replace `PROJECT_ID` with your project's id in this line:
+
+    `host: "modelserve.endpoints.PROJECT_ID.cloud.goog"`
+
+    Note that this file defines the API specifying the input `X` to be an array of arrays of floats, and output `y` to be an array of floats.  The model included in this sample code is a linear regression model with 2-dimensional inputs.
+
+1. Deploy the service endpoint:
+
+    `gcloud service-management deploy modelserve.yaml`
+
+    This step deploys a [Cloud Endpoint](https://cloud.google.com/endpoints/) service, which allows us to monitor the API usage.
+
+1. If the deploymenti s successful, get the deployment's config id from console.  It should look like `2017-08-03r0`.
+
+1. Create a GCS bucket and copy the model file over:
+
+    ```
+    gsutil mb gs://BUCKET_NAME
+    gsutil cp lr.pkl gs://BUCKET_NAME
+    ```
+
+1. Update `app.yaml`:  
+
+    - Replace `PROJECT_ID` with your project's id.
+
+    - Replace `BUCKET_NAME` with the name of the bucket you created on GCS above.
+
+    - Replace `CONFIG_ID` with the config id you got from the service endpoint deployment.
+
+1. Deploy the backend service:
+
+    `gcloud app deploy`
+
+
+1. If the deployment is successful, you can test it from the command line.  Firt go to your project's console's IAM page and create an API key `API_KEY`, and use it and your project's id in the following:
+
+    `curl -H "Content-Type: application/json" -X POST -d '{"X": [[1, 2], [5, -1], [1, 0]]}' "https://modelserve-dot-PROJECT_ID.appspot.com/predict?key=API_KEY"`
+
+    You should get the following response:
+
+    `{"y": [0.6473534912754967, -0.7187842827829021, 0.3882338314071392]}`
```

Comment:
I got the following response:

```
(gae-serve)nkash@nkash:~/GoogleCloudPlatform/ml-on-gcp/gae_serve$ curl -H "Content-Type: application/json" -X POST -d '{"X": [[1, 2], [5, -1], [1, 0]]}' "https://modelserve-dot-PROJECT_ID.appspot.com/predict?key=$GAE_SERVE_API_KEY"
<!DOCTYPE html>
<html lang=en>
  <meta charset=utf-8>
  <meta name=viewport content="initial-scale=1, minimum-scale=1, width=device-width">
  <title>Error 404 (Page not found)!!1</title>
  <style>
    *{margin:0;padding:0}html,code{font:15px/22px arial,sans-serif}html{background:#fff;color:#222;padding:15px}body{margin:7% auto 0;max-width:390px;min-height:180px;padding:30px 0 15px}* > body{background:url(//www.google.com/images/errors/robot.png) 100% 5px no-repeat;padding-right:205px}p{margin:11px 0 22px;overflow:hidden}ins{color:#777;text-decoration:none}a img{border:0}@media screen and (max-width:772px){body{background:none;margin-top:0;max-width:none;padding-right:0}}#logo{background:url(//www.google.com/images/branding/googlelogo/1x/googlelogo_color_150x54dp.png) no-repeat;margin-left:-5px}@media only screen and (min-resolution:192dpi){#logo{background:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) no-repeat 0% 0%/100% 100%;-moz-border-image:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) 0}}@media only screen and (-webkit-min-device-pixel-ratio:2){#logo{background:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) no-repeat;-webkit-background-size:100% 100%}}#logo{display:inline-block;height:54px;width:150px}
  </style>
  <a href=//www.google.com/><span id=logo aria-label=Google></span></a>
  <p><b>404.</b> <ins>That’s an error.</ins>
  <p>The requested URL was not found on this server.  <ins>That’s all we know.</ins>
```

Because of firewall, though, probably.

Reply:
Actually, the reason I was getting that error was because I forgot to replace the `PROJECT_ID` in the URL in my curl command. Works now.

Reply:
added note in README to reminder the user.